src/meltano/core/plugin_install_service.py

```diff
@@ -323,7 +317,13 @@ async def install_plugin_async(
             ),
         )
 
-        if not plugin.is_installable() or self._is_mapping(plugin):
+        is_installed = self.project.plugin_dir(plugin, "venv", make_dirs=False).exists()
```

Comment:
This is a very basic check to determine whether or not to install based on the presence of the plugin virtual environment directory. Is there anything more intelligent we can do here to satisfy the stale plugin requirement from #2089? https://github.com/meltano/meltano/issues/6416#issuecomment-1922486621 and #2651 seem relevant.

Reply:
The `VenvService` uses a fingerprint computed from the (expanded) `pip_url`:

https://github.com/meltano/meltano/blob/effdf0bf6f346002951022ef60e6e98eb8b29869/src/meltano/core/venv_service.py#L261-L272

Maybe that's helpful?

Reply:
Thanks - addressed this in https://github.com/meltano/meltano/pull/8482/commits/21cb02df2fda6a826365f9cf7346cc029b0c419a